protocol/0075-PLUP_protocol_upgrades.md

```diff
@@ -0,0 +1,166 @@
+
+
+### Multiple proposals (<a name="0075-PLUP-021" href="#0075-PLUP-021">0075-PLUP-021</a>)
+   - If multiple proposals are submitted from a validator before the block heights are reached then only the last proposal is considered
+
+## Spam (<a name="0075-PLUP-022" href="#0075-PLUP-022">0075-PLUP-022</a>) 
+   - Excessive numbers of proposals from a single validator within an epoch should be detected and rejected - (Future requirement)
+   
+## Snapshots
+   - (<a name="0075-PLUP-023" href="#0075-PLUP-023">0075-PLUP-023</a>) Post validator joining they should be immediately allowed to propose and included in the overall total count
+   - (<a name="0075-PLUP-024" href="#0075-PLUP-024">0075-PLUP-024</a>) Validators attempting to join during upgrade will be rejected if protocol versions differ
+   - (<a name="0075-PLUP-025" href="#0075-PLUP-025">0075-PLUP-025</a>) Ensure that required majority is not met when enough validators join between validator proposals and target block
+   - (<a name="0075-PLUP-026" href="#0075-PLUP-026">0075-PLUP-026</a>) Node starting from snapshot which has a proposal at a given block, ensure during replay when the block height is reached a new version is loaded and also post load an upgrade takes place at target block.
+
+## Checkpoints
+   - (<a name="0075-PLUP-027" href="#0075-PLUP-027">0075-PLUP-027</a>) Validator proposals should not be stored in the checkpoints and restored into the network
+   - (<a name="0075-PLUP-028" href="#0075-PLUP-028">0075-PLUP-028</a>) Upgrade will not occur after a post checkpoint restore until new proposals are made and block height reached
+   
+## API
+   - (<a name="0075-PLUP-029" href="#0075-PLUP-029">0075-PLUP-029</a>) An datanode API should be available to provide information on the upcoming confirmed proposal including total proposals/block details/versions
+   
+
+### Successful upgrade  (<a name="0075-PLUP-030" href="#0075-PLUP-030">0075-PLUP-030</a>) 
+  - A new release is made available, and is successfully deployed
+  - Setup a network with 5 validators running version x
+  - Have 4 validator submit request to upgrade to release >x at block height 1000
+  - At the end of block height 1000 a snapshot is taken and vega is stopped by the vegavisor
+  - All nodes are starting from the snapshot of block 1000 and the network resumes with version >x
+ 
+ 
+ ### Failing consensus (<a name="0075-PLUP-031" href="#0075-PLUP-031">0075-PLUP-031</a>)
+  - Upgrade takes place at block N. Minimum number of validators -1 restart with correct version. One validator restarts with previous version. Consensus is not achieved. Start another validator with the correct version, consensus is achieved
```

Comment:
Regardless of what own stake + delegations they have?

Reply:
Step 1:
Restart the network with a number of validators whose voting power is equal to less than two thirds. Consensus is not achieved.
Step 2:
Restart one of the validators (whose voting power would take the total voting power over the two thirds threshold) with the wrong version. Consensus should still not be achieved.
Step 3:
Restart validator from step 2 with the correct version. Consensus is achieved.